src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -236,6 +242,17 @@ internal static async Task HandleExceptionAsync(TimeoutAttribute timeout, Except
                 // Create a linked token source that will allow us to signal function cancellation
                 // (e.g. Based on TimeoutAttribute, etc.)                
                 CancellationTokenSource functionCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+                if (_drainModeManager?.IsDrainModeEnabled == true)
+                {
+                    functionCancellationTokenSource.Cancel();
+                    logger?.LogInformation($"Requesting cancellation for function invocation:{instance.Id}");
+                }
+                else
+                {
+                    _drainModeManager.RegisterTokenSource(instance.Id, functionCancellationTokenSource);
```

Comment:
We still have a problem with the potential `null` exception here (if `null`, you'll fall into this block)

Reply:
You might want to have a test with that to ensure this is covered.

Reply:
Have added a test. Found an issue as a result and have fixed it.